### PR TITLE
Fixes for the field sampling to recover convergence

### PR DESCRIPTION
**Pull request by vijaysm**
This PR changeset contains several key fixes to the sampling functional and how both the closed form analytical functions and SPH samplers are evaluated to compute element-wise averaged data for FV-FV remap comparisons. 

- Key issue that resolved the bulk of the problems was removing the usage of `np.unique` call when getting the connectivity of an element
- Secondly, the cartesian coordinates to lat/lon transformations now use more standard definitions
- In addition to the Gauss quadrature integration, we now have an optional method to evaluate cell integrals with Radial quadratures. This module requires Numba. Contributed by Dr. Xiangmin Jiao and Yipeng Li
- New tests have been added to independently evaluate convergence orders of closed form functions in order to isolate the issues in the CANGAFieldGenDriver. This can be used as a regression test going forward.
- Several performance improvements with Python multiprocessing
- Several miscellaneous updates and fixes in preprocessing routines

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Go for it and send all this to master.

File: TestIntegralConvergenceFV.py [hunk 1]
```diff
@@ -111,7 +113,7 @@ def test2(lonRAD, latRAD):
        thisAvgs2 = np.zeros(numCells)
        thisAreas = np.zeros(numCells)
        for cc in range(numCells):
-              cdex = thisCell[cc,:] - 1
+              cdex = thisCell[cc,:] - 1 ######### NP.UNIQUE BUG #######
```
Comment (kuberry) [inline TestIntegralConvergenceFV.py]: Do you want this comment here?

File: computeLocalExtremaMetrics.py [hunk 2]
```diff
@@ -144,41 +144,26 @@ def computeLocalExtremaMetrics(varConStenDex, varCon, varCoord, varS2T, varST, a
...
-       
-       if SpectralElement:
-              Lmin_1 = np.asscalar(L1Num / L1Den)
-              Lmin_2 = mt.sqrt(L2Num / L2Den)
-              Lmin_inf = LinfNum / LinfDen
-       else:
-              Lmin_1 = np.asscalar(L1Num / NT / LinfDen / LareaDen)
-              Lmin_2 = mt.sqrt(L2Num / NT) / LinfDen / LareaDen
-              Lmin_inf = LinfNum / LinfDen
+       Lmin_1 = np.asscalar(L1Num / L1Den)
```
Comment (kuberry) [inline computeLocalExtremaMetrics.py:156]: Not very important, but asscalar is deprecated, so could try using (L1Num / L1Den).item()

**Comment by kuberry** [comment]
I read all changed lines in the review (two small suggestions), but I didn't run any tests. Looks good, I'll approve.

**Review by kuberry — APPROVED** [review]

File: computeLocalExtremaMetrics.py (hunk 2 above)
Comment (vijaysm) [reply computeLocalExtremaMetrics.py:156]: Will do.

File: TestIntegralConvergenceFV.py (hunk 1 above)
Comment (vijaysm) [reply TestIntegralConvergenceFV.py]: Yes, not relevant anymore. I'll clean it up.

**Comment by vijaysm** [comment]
I'll make the minor updates and merge. I do have another commit that I need to push. Will add that as well.

**Pull request merged**